File: check_transparency.js
const fs = require('fs');
const path = require('path');

const files = [
    'public/images/hero-image2.png',
    'public/images/bottle-classic.png',
    'public/images/bottle-tropical.png'
];

files.forEach(file => {
    try {
        if (!fs.existsSync(file)) {
            console.log(`${file}: Not Found`);
            return;
        }

        const fd = fs.openSync(file, 'r');
        const buffer = Buffer.alloc(30);
        fs.readSync(fd, buffer, 0, 30, 0);
        fs.closeSync(fd);

        if (buffer.toString('hex', 0, 8) !== '89504e470d0a1a0a') {
            console.log(`${file}: Not a valid PNG`);
            return;
        }

        const colorType = buffer.readUInt8(25);
        let hasAlpha = (colorType === 6 || colorType === 4);

        console.log(`${file}: Color Type ${colorType}, Has Alpha: ${hasAlpha}`);

    } catch (e) {
        console.log(`${file}: Error - ${e.message}`);
    }
});

File: check_transparency.py
from PIL import Image
import os

files = [
    'public/images/hero-image2.png',
    'public/images/bottle-classic2.png',
    'public/images/bottle-tropical2.png'
]

def check_transparency(path):
    try:
        if not os.path.exists(path):
            print(f"{path}: Not found")
            return

        img = Image.open(path)
        if img.mode != 'RGBA':
            print(f"{path}: No Alpha Channel (Mode: {img.mode})")
            return
        
        extrema = img.getextrema()
        # extrema[3] is the alpha channel min/max
        if extrema[3][0] < 255:
             print(f"{path}: Has Transparency")
        else:
             print(f"{path}: Alpha Channel exists but is fully opaque")
            
    except Exception as e:
        print(f"{path}: Error - {e}")

for f in files:
    check_transparency(f)
